### DISCUSSION TITLE
Add a `scrollAreaProps` for Modal

### DISCUSSION THREAD

**Discussion by ShookLyngs** [mantinedev/mantine#3677] [Feature requests]
The modal component in v6 provides built-in ScrollArea, but if I'm correct, right now we cannot pass props into the ScrollArea right? My goal is to customize the style of the ScrollArea, thanks.

**Comment by rtivital** [comment]
You can customize scrollarea component with `scrollAreaComponent` prop – https://mantine.dev/core/modal/#native-scrollbars

**Reply by oiqwrwer1** [reply]
What's the best way to keep the UI behavior previously provided by Modal overflow="inside"  ?
Previously, the scrollbar was only below the header: 
![image](https://user-images.githubusercontent.com/102306502/225107387-a002a1ee-1198-44c6-8068-302b0378c6b3.png)
Now, the scrollbar is including the header and my <Tabs>content in the Modal body covers the header when scrolled:
![image](https://user-images.githubusercontent.com/102306502/225107485-4790946a-7505-4401-981c-5ee12e9b71cb.png)

**Reply by rtivital** [reply]
You can render your own header by setting `withCloseButton={false}` and removing `title` prop.

**Reply by mattaningram** [reply]
I want to have a drawer/modal where just the content scrolls but I don't see any option to turn off auto scrolling in general. Even hiding the header with `withCloseButton` and setting `Modal.NativeScrollArea` it adds `overflow-y: auto` when I actually don't want any overflow on that element.

To get just the content area to scroll (we also have a footer we added which we don't want to scroll either) we had to use some awkward styles to target the empty `div` that gets added with NativeScrollArea which has no classes or style name access so we could make everything inside grow to fill the available space.

Side note: That's a somewhat confusing prop name for hiding the entire header. I would assume it would just hide the close button so you can implement your own.
